src/components/Container/Container.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import { Courses } from '../Courses/Courses';
+import { EmptyList } from '../Courses/components/EmptyList/EmptyList';
+import { CourseInfo } from '../CourseInfo/CourseInfo';
+
+function replaceAuthorsIds(courses, authors) {
+	return courses.map((course) => {
+		if (typeof course.authors[0] !== 'object') {
+			const courseAuthors = course.authors.map((courseAuthorId) => {
+				function findAuthor(author) {
+					return author.id === courseAuthorId;
+				}
+
+				return authors.find(findAuthor);
+			});
+			course.authors = courseAuthors;
+			return course;
+		}
+		return course;
+	});
+}
+
+const Container = (props) => {
+	const courses = replaceAuthorsIds(props.courses, props.authors);
```

Comment:
[nice to have]
const courses = useMemo(() => replaceAuthorsIds(props.courses, props.authors), [props.courses, props.authors]);